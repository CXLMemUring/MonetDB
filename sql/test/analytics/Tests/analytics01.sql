start transaction;
create table analytics (aa int, bb int, cc bigint);
insert into analytics values (15, 3, 15), (3, 1, 3), (2, 1, 2), (5, 3, 5), (NULL, 2, NULL), (3, 2, 3), (4, 1, 4), (6, 3, 6), (8, 2, 8), (NULL, 4, NULL);

select percent_rank() over (partition by aa) from analytics;
select percent_rank() over (partition by aa order by aa asc) from analytics;
select percent_rank() over (partition by aa order by aa desc) from analytics;
select percent_rank() over (order by aa) from analytics;
select percent_rank() over (order by aa desc) from analytics;

select percent_rank() over (partition by bb) from analytics;
select percent_rank() over (partition by bb order by bb asc) from analytics;
select percent_rank() over (partition by bb order by bb desc) from analytics;
select percent_rank() over (order by bb) from analytics;
select percent_rank() over (order by bb desc) from analytics;

rollback;
