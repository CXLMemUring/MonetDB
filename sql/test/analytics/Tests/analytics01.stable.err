stderr of test 'analytics01` in directory 'sql/test/analytics` itself:


# 16:22:53 >  
# 16:22:53 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30581" "--set" "mapi_usock=/var/tmp/mtest-23680/.s.monetdb.30581" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 16:22:53 >  

# builtin opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 30581
# cmdline opt 	mapi_usock = /var/tmp/mtest-23680/.s.monetdb.30581
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics
# cmdline opt 	embedded_c = true
# cmdline opt 	gdk_debug = 553648138

# 16:22:53 >  
# 16:22:53 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-23680" "--port=30581"
# 16:22:53 >  


# 16:22:53 >  
# 16:22:53 >  "Done."
# 16:22:53 >  

