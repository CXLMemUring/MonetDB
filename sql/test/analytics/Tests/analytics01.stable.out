stdout of test 'analytics01` in directory 'sql/test/analytics` itself:


# 17:00:05 >  
# 17:00:05 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=39763" "--set" "mapi_usock=/var/tmp/mtest-22291/.s.monetdb.39763" "--set" "monet_prompt=" "--forcemito" "--dbpath=/home/ferreira/repositories/MonetDB-analytics/BUILD/var/MonetDB/mTests_sql_test_analytics" "--set" "embedded_c=true"
# 17:00:05 >  

# MonetDB 5 server v11.32.0
# This is an unreleased version
# Serving database 'mTests_sql_test_analytics', using 8 threads
# Compiled for x86_64-pc-linux-gnu/64bit with 128bit integers
# Found 15.492 GiB available main-memory.
# Copyright (c) 1993 - July 2008 CWI.
# Copyright (c) August 2008 - 2018 MonetDB B.V., all rights reserved
# Visit https://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://wired-142.cwi.nl:39763/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-22291/.s.monetdb.39763
# MonetDB/GIS module loaded
# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_views.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 60_wlcr.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 85_bam.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql
# MonetDB/SQL module loaded

Ready.

# 17:00:05 >  
# 17:00:05 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-22291" "--port=39763"
# 17:00:05 >  

#start transaction;
#create table analytics (aa int, bb int, cc bigint);
#insert into analytics values (15, 3, 15), (3, 1, 3), (2, 1, 2), (5, 3, 5), (NULL, 2, NULL), (3, 2, 3), (4, 1, 4), (6, 3, 6), (8, 2, 8), (NULL, 4, NULL);
[ 10	]
#select percent_rank() over (partition by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (partition by aa order by aa asc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (partition by aa order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (order by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0.2222222222	]
[ 0.3333333333	]
[ 0.3333333333	]
[ 0.5555555556	]
[ 0.6666666667	]
[ 0.7777777778	]
[ 0.8888888889	]
[ 1	]
#select percent_rank() over (order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0.1111111111	]
[ 0.2222222222	]
[ 0.3333333333	]
[ 0.4444444444	]
[ 0.5555555556	]
[ 0.5555555556	]
[ 0.7777777778	]
[ 0.8888888889	]
[ 0.8888888889	]
#select percent_rank() over (partition by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (partition by bb order by bb asc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (partition by bb order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select percent_rank() over (order by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0.3333333333	]
[ 0.3333333333	]
[ 0.3333333333	]
[ 0.6666666667	]
[ 0.6666666667	]
[ 0.6666666667	]
[ 1	]
#select percent_rank() over (order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0.1111111111	]
[ 0.1111111111	]
[ 0.1111111111	]
[ 0.4444444444	]
[ 0.4444444444	]
[ 0.4444444444	]
[ 0.7777777778	]
[ 0.7777777778	]
[ 0.7777777778	]
#select cume_dist() over (partition by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.1	]
[ 0.2	]
[ 0.3	]
[ 0.4	]
[ 0.5	]
[ 0.7	]
[ 0.7	]
[ 0.8	]
[ 1	]
[ 1	]
#select cume_dist() over (partition by aa order by aa asc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.2	]
[ 0.2	]
[ 0.3	]
[ 0.5	]
[ 0.5	]
[ 0.6	]
[ 0.7	]
[ 0.8	]
[ 0.9	]
[ 1	]
#select cume_dist() over (partition by aa order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.1	]
[ 0.2	]
[ 0.3	]
[ 0.4	]
[ 0.5	]
[ 0.7	]
[ 0.7	]
[ 0.8	]
[ 1	]
[ 1	]
#select cume_dist() over (order by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select cume_dist() over (order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select cume_dist() over (partition by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.1	]
[ 0.4	]
[ 0.4	]
[ 0.4	]
[ 0.7	]
[ 0.7	]
[ 0.7	]
[ 1	]
[ 1	]
[ 1	]
#select cume_dist() over (partition by bb order by bb asc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.3	]
[ 0.3	]
[ 0.3	]
[ 0.6	]
[ 0.6	]
[ 0.6	]
[ 0.9	]
[ 0.9	]
[ 0.9	]
[ 1	]
#select cume_dist() over (partition by bb order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0.1	]
[ 0.4	]
[ 0.4	]
[ 0.4	]
[ 0.7	]
[ 0.7	]
[ 0.7	]
[ 1	]
[ 1	]
[ 1	]
#select cume_dist() over (order by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select cume_dist() over (order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over (partition by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over (partition by aa order by aa asc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over (partition by aa order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over (order by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over (order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(3) over (partition by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
#select ntile(3) over (partition by bb order by bb asc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
#select ntile(3) over (partition by bb order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
[ 1	]
[ 2	]
[ 3	]
#select ntile(3) over (order by bb) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 2	]
[ 2	]
[ 2	]
[ 3	]
[ 3	]
[ 3	]
#select ntile(3) over (order by bb desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 2	]
[ 2	]
[ 2	]
[ 3	]
[ 3	]
[ 3	]
#select ntile(10) over (partition by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 2	]
[ 1	]
[ 1	]
[ 2	]
#select ntile(10) over (partition by aa order by aa asc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 2	]
[ 1	]
[ 1	]
[ 2	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(10) over (partition by aa order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 2	]
[ 1	]
[ 1	]
[ 2	]
#select ntile(10) over (order by aa) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 2 # length
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
#select ntile(10) over (order by aa desc) from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 2 # length
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
#select percent_rank() over () from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
[ 0	]
#select cume_dist() over () from analytics;
% sys.L4 # table_name
% L4 # name
% double # type
% 24 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(1) over () from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
#select ntile(2) over () from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 1	]
[ 2	]
[ 2	]
[ 2	]
[ 2	]
[ 2	]
#select ntile(5) over () from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 1 # length
[ 1	]
[ 1	]
[ 2	]
[ 2	]
[ 3	]
[ 3	]
[ 4	]
[ 4	]
[ 5	]
[ 5	]
#select ntile(11) over () from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 2 # length
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
#select ntile(100) over () from analytics;
% sys.L4 # table_name
% L4 # name
% tinyint # type
% 2 # length
[ 1	]
[ 2	]
[ 3	]
[ 4	]
[ 5	]
[ 6	]
[ 7	]
[ 8	]
[ 9	]
[ 10	]
#rollback;

# 17:00:05 >  
# 17:00:05 >  "Done."
# 17:00:05 >  

